src/PointwiseFunctions/Xcts/LongitudinalOperator.cpp

```diff
@@ -0,0 +1,89 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/Elasticity/PotentialEnergy.hpp"
+
+#include <algorithm>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Xcts {
+
+/// \cond
+template <typename DataType>
+void longitudinal_operator(const gsl::not_null<tnsr::II<DataType, 3>*> result,
+                           const tnsr::ii<DataType, 3>& strain,
+                           const tnsr::II<DataType, 3>& inv_metric) noexcept {
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j <= i; ++j) {
+      // Unroll first iteration of the loop over `k` to avoid filling the result
+      // with zero initially
+      result->get(i, j) = 2. *
+                          (inv_metric.get(i, 0) * inv_metric.get(j, 0) -
+                           inv_metric.get(i, j) * get<0, 0>(inv_metric) / 3.) *
+                          get<0, 0>(strain);
+      for (size_t l = 1; l < 3; ++l) {
+        result->get(i, j) +=
+            2. *
+            (inv_metric.get(i, 0) * inv_metric.get(j, l) -
+             inv_metric.get(i, j) * inv_metric.get(0, l) / 3.) *
+            strain.get(0, l);
+      }
+      for (size_t k = 1; k < 3; ++k) {
+        for (size_t l = 0; l < 3; ++l) {
+          const auto projection =
+              2. * (inv_metric.get(i, k) * inv_metric.get(j, l) -
+                    inv_metric.get(i, j) * inv_metric.get(k, l) / 3.);
```

Comment:
Math is correct, but I was thrown off by the unusual '/3' at the end.  Suggest to write
```suggestion
              2. * inv_metric.get(i, k) * inv_metric.get(j, l) 
              -(2./3.) * inv_metric.get(i, j) * inv_metric.get(k, l);
```
to bring code into closer proximity to how formulae are written in papers (of course, also in the loop-unrolled copies above)